### ISSUE TITLE
parsing monarch ontology with fastobo fails

### ISSUE THREAD

**Issue by sierra-moxon** [monarch-initiative/monarch-ontology#39]
```
    def parse_from(self, handle, threads=None):
        # Load the OBO graph into a syntax tree using fastobo
>       doc = fastobo.load_graph(handle).compact_ids()
E         File "<stdin>", line 1
E           https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:9462
E                                                              ^
E       SyntaxError: remaining input
```

Possible to update the id here to not have that `#!` ? 
from @cmungall 
Could be that this needs to be fixed upstream in prefixcommons. 

Ran into this in using oaklib to parse and walk this ontology in hopes of better assigning biolink-model categories in KGX and for monarch-ingest.  In parallel, I'll try another parsing implementation in oaklib.

**Comment by matentzn** [comment]
I think it is not worth trying to fix monarch.owl. We need to make a concerted push to PHENIO instead. But this issue will still have to be dealt with!

In the meantime: best not (ever) use monarch.owl in OBO format. use obographs.

**Comment by sierra-moxon** [comment]
I am actually getting this error from this file (in obo json format): https://ci.monarchinitiative.org/job/monarch-ontology-json-sri/lastSuccessfulBuild/artifact/build/monarch-ontology-final.json

**Comment by matentzn** [comment]
That is then IMO a bug with `fastobo`.. This is a valid IRI - its just not a valid OBO id! @althonos, do you agree?

**Comment by althonos** [comment]
Yes, looks like a bug in the syntax, the IRI is definitely valid. When trying to force parsing an IRI using the current PEG grammar it will not parse the fragment part even though it should. The `!` is always a problem in OBO because it's used as a comment character, but is valid in IRI context... 

In any case though, the `fastobo-graphs` implementation shouldn't attempt to parse IRIs as OBO IDs because they are not going to be in escaped form in the JSON I suppose. I'll have a look when I can (currently in the final stretch of an unrelated manuscript)!
